### PR TITLE
SONAR-7724 WS api/permissions/remove_project_creator_from_template

### PR DESCRIPTION
**Pull request by teryk**
review only the last commit about remove_project_creator_from_template

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/AddProjectCreatorToTemplateAction.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  private void updateTemplatePermission(DbSession dbSession, PermissionTemplatePermissionDto templatePermission) {
+    PermissionTemplatePermissionDto targetTemplatePermission = templatePermission
+      .setUpdatedAt(system.now())
+      .setWithProjectCreator(true);
+    dbClient.permissionTemplatePermissionDao().update(dbSession, targetTemplatePermission);
+    dbSession.commit();
+  }
+
+  private AddProjectCreatorToTemplateWsRequest toWsRequest(Request request) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/AddProjectCreatorToTemplateAction.java]: MAJOR Make "toWsRequest" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/AddProjectCreatorToTemplateAction.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    checkGlobalAdminUser(userSession);
+    doHandle(toWsRequest(request));
+    response.noContent();
+  }
+
+  private void doHandle(AddProjectCreatorToTemplateWsRequest request) {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      PermissionTemplateDto template = dependenciesFinder.getTemplate(dbSession, WsTemplateRef.newTemplateRef(request.getTemplateId(), request.getTemplateName()));
+      Optional<PermissionTemplatePermissionDto> templatePermission = dbClient.permissionTemplatePermissionDao().selectByPermissionAndTemplateId(dbSession, request.getPermission(), template.getId());
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/AddProjectCreatorToTemplateAction.java]: MINOR Split this 198 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplatePermissionDao.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return dto;
+  }
+
+  public PermissionTemplatePermissionDto update(DbSession dbSession, PermissionTemplatePermissionDto templatePermissionDto) {
+    requireNonNull(templatePermissionDto.getId());
+    mapper(dbSession).update(templatePermissionDto);
+    return templatePermissionDto;
+  }
+
+  private PermissionTemplatePermissionMapper mapper(DbSession dbSession) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/template/PermissionTemplatePermissionDao.java]: MAJOR Make "mapper" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 3 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR org.sonarsource.sonarqube:sonar-db:src/main/java/org/sonar/db/permission/template: Add a 'package-info.java' file to document the 'template' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR org.sonarsource.sonarqube:sonar-db:src/main/java/org/sonar/db/permission/template: Add a 'package-info.java' file to document the 'template' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
LGTM

**Pull request merged**